Velocity4LumiContainers/README.md

```diff
@@ -0,0 +1,122 @@
+  - `velocity build rocm@6.2.4 fakeGpu libcxi libfabric@2.3.0 mpich@4.2.3 xcclPlugin@0.3 tests -v`
+- future
+  - `velocity build rocm@7.0.2 fakeGpu libcxi libfabric@2.3.0 mpich@4.3.2 xcclPlugin@0.3 tests -v`
+
+To build a pytorch container with the `latest` version of the lumi container one could do:
+- `velocity build opensuse@15.5 rocm@6.2.4 fakeGpu libcxi libfabric@2.3.0 mpich@4.2.3 xcclPlugin@0.3 tests pythonMiniforge python@3.12 pythonPytorch@2.7.1 -v`
+
+This will build the container with all the Lumi communication bits needed as well as install a conda venv with Pytorch pre-installed. 
+
+# Versions
+
+Some packages do not have versions but use git hashes. 
+`Velocity` currently does not support version numbers that are not in a typical Major.Minor(.Patch) format. 
+
+- xcclPlugin
+  - 0.1 --> stable `aws-ofi-rccl` as is included in the standard Lumi containers
+  - 0.2 --> latest `aws-ofi-rccl`
+  - 0.3 --> `open-ofi-xccl` v1.14.x-xxx; first version that supports ROCm.
+
+- libcxi
+  - 0.1 --> stable as is included in the standard Lumi containers. Git hashes for: `libcxi`, `cxi_driver` and `cassini_headers`
+
+- libfabric
+  - 1.9 --> stable as is included in the standard Lumi containers based on git hash
+  - 2.3 --> enables the lnx provider to combine multiple providers
+
+- fakeGPU
+  - 0.1 --> version to fake LUMI GPUs
+
+- tests
+  - 0.1 --> installs both OSU benchmarks and RCCL tests
+
+- packages
+  - 1.0 --> various normal packages that are required for the other installations (gcc, cmake, autoconf, tar, patch git etc.)
+
+# Results
+
+## Pros
+
+- Easy to setup
+- Creates sif containers from predefined setup scripts
+- Adding new libraries etc. is easy - it's basically just bash scripts for your specific linux distro (e.g., zypper with opensuse).
+- Dependency resolution seems to work well
+- It is easy to combine different versions and set dependencies between versions of different packages. 
+- Resulting containers work on Lumi (as long as the packages are compatible with Lumi)
+- Network speeds are similar to other open source containers
+
+## Cons
+- Cannot make versions with name "X.X.X_dev" or "working" etc. This would be handy if it's an experimental version from e.g. a git commit
+- The error messages are limited; e.g., if your version isn't some sort of number it'll just say "No available build".
+- Error message if using `=` when setting envar with `!envar` is cryptic.
+- Folders cannot contain `_`
+- Mapping values are not allowed (?) i.e. `libfabric@2.0.0:`
+- `Python 3.10` becomes `Python 3.1`
+- The caching is currently only stable in the `develop` branch due to issues with the hashes.
+
+
+## Nice to have features
+- Top level folder that groups packages. e.g.: Python related stuff, communication, different MPI (MPICH, OpenMPI) etc.
+- Configuration File --> allow to specify a full container toolchain for a config. e.g., `velocity build lumi_latest` should build a container according to the config file. 
```

Comment:
Again, I think this might be at least partially possible by specifying the `system` option when declaring dependencies. You would probably still need to manually specify version numbers for packages to be considered "lumi_latest", though.